guardian/recipes-backend
# Infrastructure for serving full REST APIs & curation update endpoint

## What does this change?

- Adds API Gateway going through to a single, backend lambda via proxy integration with mandatory API keys
- POST endpoint to handle incoming curation data

## How to test

When deployed to CODE, take a copy of `curation.json`:
```bash
curl https://recipes.code.dev-guardianapis.com/curation.json > curation.json
```
Verify that it looks like the proper curation document

Now post another json document:
```bash
curl -X POST https://recipes.code.dev-guardianapis.com/api/curation -d@file.json --header 'Content-Type: application/json' --header 'x-api-key: {your-api-key}'
```

You should get a 200 response.

Now read it back:
```bash
curl https://recipes.code.dev-guardianapis.com/curation.json
```

The CDN is set up to reply with stale content during refresh, so you might have to do this a few times in order to get the new data. A cache flush is performed by the lambda, as you can see from the logs.

## How can we measure success?

Curation updates going from minutes/hours (i.e. Ben has to craft the json and send it to us, we have to manually upload) to seconds (Ben sends it directs and tests it)

## Have we considered potential risks?

This is the first POST endpoint on the service. It's protected by an API key which is sufficient for now. We can look into key rotation down the line as the feast app gets more mature
